### PR TITLE
Fix atmosphere rendering when globe is hidden

### PR DESCRIPTION
**Pull request by ggetz**
Fixes https://github.com/CesiumGS/cesium/issues/11266

There was a bug that prevented the sky atmosphere to be shown when the globe was initially disabled, whether through setting `globe: false` in the viewer options, or by hiding the globe with `globe.show = false`. This was a quick fix in `Scene.js`.

However, it does make sense that the atmosphere would be hidden if the globe is initially set to `false`. Not every use case will be with Google's photogrammetric 3D Tiles or other 3D Tiles globes. So I did add some logic that if `globe: false` is specified in the Viewer or CesiumWidget options that the atmosphere is hidden. However it can be show by either setting `scene.skyAtmosphere.show = true` or by passing a custom `SkyAtmosphere` during initialization.

Despite the bug, this is technically breaking behavior so I could see an argument for listing a note in `CHANGES.md` under the `Breaking changes` section.

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @ggetz!
* :heavy_check_mark: Signed CLA found.
* :grey_question: Unit tests were not updated.
    * Make sure you've [updated tests](https://github.com/CesiumGS/cesium/tree/main/Documentation/Contributors/TestingGuide) to reflect your changes, added tests for any new code, and ran the code coverage tool.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.

**Review by jjhembd — APPROVED** [review]
Thanks @ggetz. I think the change makes sense, and it works on my end. In both the [3D Tiles Next S2 Globe](https://sandcastle.cesium.com/?src=3D%20Tiles%20Next%20S2%20Globe.html) and [Google Photorealistic 3D Tiles](https://sandcastle.cesium.com/?src=Google%20Photorealistic%203D%20Tiles.html&label=All) sandcastles, a sky atmosphere is rendered if I do either of the following:
- Set `scene.globe.show = false` AFTER initializing a default `Viewer`, OR
- Initialize a `Viewer` with `globe: false`, then set `scene.skyAtmosphere.show = true`

This feels more like a bug fix to me, rather than a breaking change. The change to existing behavior would be if a user:
1. Initialized a viewer
2. Turned off the globe later.

With this PR, the atmosphere will stay on instead of disappearing. But I don't think anyone would have expected the atmosphere to turn off without a deliberate change to `scene.skyAtmosphere.show`.

**Comment by ggetz** [comment]
Thanks @jjhembd! Thanks for the review. I think we're agreement here then. OK to merge once I merge in `main` then?

**Pull request merged**
